cmd/kube-apiserver/app/server.go

```diff
@@ -527,60 +549,37 @@ func parseRuntimeConfig(s *options.APIServer) (map[string]genericapiserver.APIGr
-			Disable: true,
+	for version, enablementString := range versionToEnablementString {
+		enableVersion := getRuntimeConfigValue(s, enablementString, !disableAllAPIs)
+		if !enableVersion {
+			resourceEnablementConfig.DisableVersions(version)
 		}
 	}
 
-	// "extensions/v1beta1={true|false} allows users to enable/disable the extensions API.
-	// This takes preference over api/all, if specified.
-	disableExtensions := disableAllAPIs
-	extensionsGroupVersion := "extensions/v1beta1"
-	// TODO: Make this a loop over all group/versions when there are more of them.
-	disableExtensions = !getRuntimeConfigValue(s, extensionsGroupVersion, !disableExtensions)
-	if disableExtensions {
-		apiGroupVersionOverrides[extensionsGroupVersion] = genericapiserver.APIGroupVersionOverride{
-			Disable: true,
+	for key := range s.RuntimeConfig {
+		tokens := strings.Split(key, "/")
+		if len(tokens) != 3 {
+			continue
 		}
-	}
 
-	disableAutoscaling := disableAllAPIs
-	autoscalingGroupVersion := "autoscaling/v1"
-	disableAutoscaling = !getRuntimeConfigValue(s, autoscalingGroupVersion, !disableAutoscaling)
-	if disableAutoscaling {
-		apiGroupVersionOverrides[autoscalingGroupVersion] = genericapiserver.APIGroupVersionOverride{
-			Disable: true,
-		}
-	}
-	disableBatch := disableAllAPIs
-	batchGroupVersion := "batch/v1"
-	disableBatch = !getRuntimeConfigValue(s, batchGroupVersion, !disableBatch)
-	if disableBatch {
-		apiGroupVersionOverrides[batchGroupVersion] = genericapiserver.APIGroupVersionOverride{
-			Disable: true,
-		}
-	}
+		switch {
+		case strings.HasPrefix(key, extensionsGroupVersionString+"/"):
+			if !resourceEnablementConfig.VersionEnabled(extensionsGroupVersion) {
+				return nil, fmt.Errorf("%v is disabled, you cannot its resources individually", extensionsGroupVersion)
+			}
 
-	for key := range s.RuntimeConfig {
-		if strings.HasPrefix(key, v1GroupVersion+"/") {
-			return nil, fmt.Errorf("api/v1 resources cannot be enabled/disabled individually")
-		} else if strings.HasPrefix(key, extensionsGroupVersion+"/") {
-			resource := strings.TrimPrefix(key, extensionsGroupVersion+"/")
-
-			apiGroupVersionOverride := apiGroupVersionOverrides[extensionsGroupVersion]
-			if apiGroupVersionOverride.ResourceOverrides == nil {
-				apiGroupVersionOverride.ResourceOverrides = map[string]bool{}
+			resource := strings.TrimPrefix(key, extensionsGroupVersionString+"/")
+			if getRuntimeConfigValue(s, key, false) {
+				resourceEnablementConfig.EnableResources(extensionsGroupVersion.WithResource(resource))
+			} else {
+				resourceEnablementConfig.DisableResources(extensionsGroupVersion.WithResource(resource))
 			}
-			apiGroupVersionOverride.ResourceOverrides[resource] = getRuntimeConfigValue(s, key, false)
-			apiGroupVersionOverrides[extensionsGroupVersion] = apiGroupVersionOverride
+
+		default:
+			return nil, fmt.Errorf("%v resources cannot be enabled/disabled individually", key)
```

Comment:
todo to honor enablement flags in all groups and make this a loop over `versionToEnablementString`?